src/viam/examples/modules/complex/base/impl.cpp

```diff
@@ -63,18 +64,26 @@ bool MyBase::is_moving() {
     return left_->is_moving() || right_->is_moving();
 }
 
-grpc::StatusCode MyBase::stop(const AttributeMap& extra) {
-    auto left_stop = left_->stop(extra);
-    auto right_stop = right_->stop(extra);
+void MyBase::stop(const AttributeMap& extra) {
+    std::exception* e;
 
-    // Return first of any non-ok error code received from motors.
-    if (left_stop != grpc::StatusCode::OK) {
-        return left_stop;
+    // make sure we try to stop both motors, even if the first fails.
+    try {
+        left_->stop(extra);
+    } catch (const std::exception& err) {
+        *e = err;
```

Comment:
At first, I wondered why this compiled, since `err` is const and `e` non. Then I realized that this is actually invoking the copy assignment operator. But the object it is copying into doesn't exist! There is no object allocated at `*e`, since `e` is uninitialized. But, even if it was, this would end up slicing the object, since it would invoke the copy assignment operator for `std::exception`, but the object referred to by `err` probably is a derived class.

Reply:
ooh thanks for catching this! Amended to a simpler, more straightforward approach to throwing here.